server/server.go

```diff
@@ -1433,6 +1435,10 @@ func (s *StanServer) createNatsClientConn(name string) (*nats.Conn, error) {
 	ncOpts.Password = s.natsOpts.Password
 	ncOpts.Token = s.natsOpts.Authorization
 
+	if s.opts.NATSServerCredentials != "" {
```

Comment:
Let's check that this is not already set with possibly the option passed programmatically with s.opts.NATSClientOpts (for people embedding the server). So maybe move this before line `for _, o := range s.opts.NATSClientOpts {` so that these options take precedence over the config file?